// @wordpress/components/build/validated-form-controls/components/checkbox-control.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ValidatedCheckboxControl = void 0;
var _compose = require("@wordpress/compose");
var _element = require("@wordpress/element");
var _controlWithError = require("../control-with-error");
var _checkboxControl = _interopRequireDefault(require("../../checkbox-control"));
var _jsxRuntime = require("react/jsx-runtime");
/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

const UnforwardedValidatedCheckboxControl = ({
  required,
  customValidator,
  onChange,
  markWhenOptional,
  ...restProps
}, forwardedRef) => {
  const validityTargetRef = (0, _element.useRef)(null);
  const mergedRefs = (0, _compose.useMergeRefs)([forwardedRef, validityTargetRef]);
  const valueRef = (0, _element.useRef)(restProps.checked);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_controlWithError.ControlWithError, {
    required: required,
    markWhenOptional: markWhenOptional,
    ref: mergedRefs,
    customValidator: () => {
      return customValidator?.(valueRef.current);
    },
    getValidityTarget: () => validityTargetRef.current?.querySelector('input[type="checkbox"]'),
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_checkboxControl.default, {
      __nextHasNoMarginBottom: true,
      onChange: value => {
        valueRef.current = value;
        onChange?.(value);
      }
      // TODO: Upstream limitation - CheckboxControl doesn't support uncontrolled mode, visually.
      ,
      ...restProps
    })
  });
};
const ValidatedCheckboxControl = exports.ValidatedCheckboxControl = (0, _element.forwardRef)(UnforwardedValidatedCheckboxControl);
