// @wordpress/components/build/validated-form-controls/components/input-control.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ValidatedInputControl = void 0;
var _element = require("@wordpress/element");
var _compose = require("@wordpress/compose");
var _controlWithError = require("../control-with-error");
var _inputControl = _interopRequireDefault(require("../../input-control"));
var _jsxRuntime = require("react/jsx-runtime");
/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

const UnforwardedValidatedInputControl = ({
  required,
  customValidator,
  onChange,
  markWhenOptional,
  ...restProps
}, forwardedRef) => {
  const validityTargetRef = (0, _element.useRef)(null);
  const mergedRefs = (0, _compose.useMergeRefs)([forwardedRef, validityTargetRef]);
  const valueRef = (0, _element.useRef)(restProps.value);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_controlWithError.ControlWithError, {
    required: required,
    markWhenOptional: markWhenOptional,
    customValidator: () => {
      return customValidator?.(valueRef.current);
    },
    getValidityTarget: () => validityTargetRef.current,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_inputControl.default, {
      __next40pxDefaultSize: true,
      ref: mergedRefs,
      onChange: (value, ...args) => {
        valueRef.current = value;
        onChange?.(value, ...args);
      },
      ...restProps
    })
  });
};
const ValidatedInputControl = exports.ValidatedInputControl = (0, _element.forwardRef)(UnforwardedValidatedInputControl);
