// @wordpress/components/build/validated-form-controls/components/number-control.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ValidatedNumberControl = void 0;
var _element = require("@wordpress/element");
var _compose = require("@wordpress/compose");
var _controlWithError = require("../control-with-error");
var _numberControl = _interopRequireDefault(require("../../number-control"));
var _jsxRuntime = require("react/jsx-runtime");
/**
 * WordPress dependencies
 */

/**
 * Internal dependencies
 */

const UnforwardedValidatedNumberControl = ({
  required,
  customValidator,
  onChange,
  markWhenOptional,
  ...restProps
}, forwardedRef) => {
  const validityTargetRef = (0, _element.useRef)(null);
  const mergedRefs = (0, _compose.useMergeRefs)([forwardedRef, validityTargetRef]);
  const valueRef = (0, _element.useRef)(restProps.value);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_controlWithError.ControlWithError, {
    required: required,
    markWhenOptional: markWhenOptional,
    customValidator: () => {
      return customValidator?.(valueRef.current);
    },
    getValidityTarget: () => validityTargetRef.current,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_numberControl.default, {
      __next40pxDefaultSize: true,
      ref: mergedRefs
      // TODO: Upstream limitation - When form is submitted when value is undefined, it will
      // automatically set a clamped value (as defined by `min` attribute, so 0 by default).
      ,
      onChange: (value, ...args) => {
        valueRef.current = value;
        onChange?.(value, ...args);
      },
      ...restProps
    })
  });
};
const ValidatedNumberControl = exports.ValidatedNumberControl = (0, _element.forwardRef)(UnforwardedValidatedNumberControl);
